### PR TITLE
[REVIEW] Add JNI and java bindings for matches_re [skip ci]

### PR DESCRIPTION
**Pull request by kuhushukla**
Allow bindings to do string match for regex patterns.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 1]
```diff
@@ -1804,6 +1804,14 @@ public ColumnVector stringContains(Scalar compString) {
     }
   }
 
+  public ColumnVector matchesRe(String pattern) {
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Can we have some javadocs explaining this?  REs are complicated and just pattern is not much of an explanation.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 2]
```diff
@@ -1804,6 +1804,14 @@ public ColumnVector stringContains(Scalar compString) {
...
+    assert type == DType.STRING : "column type must be a String";
+    assert pattern != null : "pattern scalar may not be null";
+    assert pattern.isEmpty() == false : "pattern string scalar may not be empty";
+    try (DevicePrediction prediction = new DevicePrediction(predictSizeFor(DType.BOOL8), "endsWith")) {
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: nit: this is not endsWith

File: java/src/main/native/src/ColumnVectorJni.cpp [hunk 3]
```diff
@@ -648,6 +649,22 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_stringContains(JNIEnv *
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_matchesRe(JNIEnv *env, jobject j_object,
+                                                                    jlong j_view_handle, jstring patternObj) {
+    JNI_NULL_CHECK(env, j_view_handle, "column is null", false);
+    JNI_NULL_CHECK(env, patternObj, "comparison string scalar is null", false);
```
Comment (revans2) [inline java/src/main/native/src/ColumnVectorJni.cpp]: nit: it is, but it isn't a scalar

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java (hunk 1 above)
Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Sorry - missed that. Addressing in the next version.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 4]
```diff
@@ -1804,6 +1804,30 @@ public ColumnVector stringContains(Scalar compString) {
     }
   }
 
+  /**
+   * Returns a boolean ColumnVector identifying rows which
+   * matching the given regex pattern but only at the beginning the string.
```
Comment (nartal1) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: matching -> matches
beginning the string-> beginning of the string.

**Comment by nartal1** [comment]
Just couple of nits. Rest all looks good.

**Comment by kuhushukla** [comment]
> Just couple of nits. Rest all looks good.

Thanks @nartal1 , addressed comments

**Comment by revans2** [comment]
@kkraus14 would you please merge this one in.

**Pull request merged**